base/errorshow.jl

```diff
@@ -1104,7 +1104,23 @@ function fielderror_hint_handler(io, exc)
     end
 end
 
-Experimental.register_error_hint(fielderror_hint_handler, FieldError)
+Experimental.register_error_hint(fielderror_dict_hint_handler, FieldError)
+
+function fielderror_listfields_hint_handler(io, exc)
+    fields = fieldnames(exc.type)
+    println(io, ", available fields: $(join(map(k -> "`$k`", fields), ", "))")
```

Comment:
We should special case the empty case:

```julia-repl
julia> nothing.bar
ERROR: FieldError: type Nothing has no field `bar`, available fields: 

Stacktrace:
 [1] getproperty(x::Nothing, f::Symbol)
   @ Base ./Base.jl:49
 [2] top-level scope
   @ REPL[1]:1
```

```suggestion
    fields = fieldnames(exc.type)
    isempty(fields)
        print(io, "; $(exc.type) has no fields.")
    end
        print(io, ", available fields:\n$(join(map(k -> "`$k`", fields), ", "))")
    end
```

Also, rather than printing a blank line after the hint, I'd put the field names on the next line.